src/puppetlabs/puppetdb/cli/util.clj

```diff
@@ -0,0 +1,41 @@
+(ns puppetlabs.puppetdb.cli.util
+  "As this namespace is required by both the tk and non-tk subcommands,
+  it must remain very lightweight, so that subcommands like
+  \"version\" aren't slowed down by loading the entire logging
+  subsystem or trapperkeeper, etc."
+  (:require
+   [puppetlabs.i18n.core :refer [trs]]))
+
+(def err-exit-status 1)
+
+;; Testing hook
+(defn java-version [] (System/getProperty "java.version"))
+
+(defn jdk-support-status [version]
+  "Returns :official, :tested, or :unknown, or :no."
+  (cond
+    (re-matches #"1.[123456]($|(\..*))" version) :no
+    (re-matches #"1.7($|(\..*))" version) :unknown
+    (re-matches #"1.8($|(\..*))*" version) :official
+    (re-matches #"1.10($|(\..*))*" version) :tested
```

Comment:
I believe they switched to using the java version with Java 10, so instead of `1.10.2` it's `10.0.2`. Maybe that's just in the cli --version flag. I'll check on that.

Reply:
```
Java HotSpot(TM) 64-Bit Server VM 10.0.2+13

user=> (System/getProperty "java.version")
"10.0.2"
user=>
```